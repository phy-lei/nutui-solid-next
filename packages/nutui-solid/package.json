{
  "name": "nutui-solid",
  "version": "1.0.0",
  "description": "",
  "author": "",
  "license": "ISC",
  "keywords": [],
  "main": "dist/nutui.solid.umd.js",
  "module": "dist/nutui.solid.es.js",
  "typings": "dist/types/index.d.ts",
  "files": [
    "dist"
  ],
  "scripts": {
    "prepare": "node scripts/generate-nutui.cjs",
    "generate:themes": "node scripts/generate-themes.cjs",
    "dev": "pnpm dev-components && pnpm build-css",
    "build": "pnpm build-components && pnpm build-css",
    "dev-components": "vite build --watch",
    "build-components": "vite build",
    "build-css": "pnpm generate:themes && vite build -c vite.config.build.css.ts"
  },
  "dependencies": {
    "classnames": "^2.5.1"
  },
  "devDependencies": {
    "@babel/core": "^7.23.3",
    "@rollup/plugin-commonjs": "^25.0.7",
    "@rollup/plugin-node-resolve": "^15.2.3",
    "@types/babel__core": "^7.20.4",
    "@types/jest": "^29.5.11",
    "autoprefixer": "^10.4.19",
    "babel-loader": "^9.1.3",
    "babel-plugin-transform-solid-jsx": "4.0.0-beta.85",
    "fs-extra": "^11.2.0",
    "jest": "^29.7.0",
    "merge-anything": "^5.1.7",
    "postcss-import": "^16.1.0",
    "rollup": "^4.9.5",
    "rollup-plugin-node-externals": "^4.0.0",
    "rollup-plugin-ts": "^3.4.5",
    "sass": "^1.77.5",
    "solid-refresh": "^0.6.3",
    "vite": "^5.1.3",
    "vite-plugin-dts": "3.6.4",
    "vitefu": "^0.2.5"
  }
}
