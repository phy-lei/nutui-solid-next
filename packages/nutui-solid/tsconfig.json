{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "baseUrl": ".",
    "jsx": "preserve",
    "jsxImportSource": "solid-js",
    "types": ["vite-plugin-pwa/info"],
    "paths": {
      "@/*": ["src/*"],
      "nutui-solid": ["./dist"]
    },
    "resolveJsonModule": true,
    "esModuleInterop": true
  },
  "include": ["**/*.ts", "**/*.tsx"]
}
