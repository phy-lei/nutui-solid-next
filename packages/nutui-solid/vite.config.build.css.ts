import { defineConfig } from 'vite'

export default defineConfig({
  build: {
    minify: true,
    lib: {
      entry: './dist/styles/themes/default.scss',
      formats: ['es'],
      name: 'style',
      fileName: 'style'
    },
    emptyOutDir: false
  }
})
