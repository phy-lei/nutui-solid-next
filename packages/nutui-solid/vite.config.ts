import { defineConfig } from 'vite'
import dts from 'vite-plugin-dts'
import fse from 'fs-extra'
import path from 'path'
import config from './package.json'
import solidPlugin from './plugin/vite-plugin-solid'

const banner = `/*!
* ${config.name} v${config.version} ${new Date()}
* (c) 2023 @jdf2e.
* Released under the MIT License.
*/`

const { resolve } = path

let fileStr = `@import "@/styles/variables.scss";`
const projectID = process.env.VITE_APP_PROJECT_ID
if (projectID) {
  fileStr = `@import '@/styles/variables-${projectID}.scss';`
}
// https://vitejs.dev/config/
export default defineConfig({
  resolve: {
    alias: [{ find: '@', replacement: resolve(__dirname, './src') }],
  },
  plugins: [
    solidPlugin(),
    dts({
      outDir: 'dist/types',
      clearPureImport: false,
      include: [
        "src/components/nutui.solid.build.ts",
        "src/components/**/*.ts",
        "src/components/**/*.tsx",
      ],
      exclude: [
        'node_modules/**',
        'src/components/**/demos/**',
        'src/**/*.spec.tsx',
        'plugin/**'
      ],
      beforeWriteFile(filePath, content) {
        return { filePath: filePath.replace('src/', ''), content }
      },
      afterBuild: () => {
        fse
          .readFile('./dist/types/nutui.solid.build.d.ts', 'utf-8')
          .then((data: string) => {
            fse.remove('./dist/types/nutui.solid.build.d.ts')
            fse.outputFile(
              './dist/types/index.d.ts',
              `${data.replace(/\.\.\//g, './')}`
            )
          })
      },
    }),
  ],
  build: {
    minify: false,
    emptyOutDir: true,
    rollupOptions: {
      // 请确保外部化那些你的库中不需要的依赖
      external: ['solid-js'],
      output: [
        {
          banner,
          globals: {
            solidJs: 'solidJs',
          },
          format: 'es',
          entryFileNames: 'nutui.solid.es.js',
        },
        {
          banner,
          globals: {
            solidJs: 'solidJs',
          },
          name: 'nutui.solid',
          format: 'umd',
          entryFileNames: 'nutui.solid.umd.js',
        },
      ],
    },
    lib: {
      entry: 'src/components/nutui.solid.build.ts',
    },
  },
})
