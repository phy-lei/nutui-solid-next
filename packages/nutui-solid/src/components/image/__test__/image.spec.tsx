import { test, expect } from "vitest"
import { fireEvent, render } from "@solidjs/testing-library"
import { Image } from 'nutui-solid'

const url = '//img10.360buyimg.com/ling/jfs/t1/181258/24/10385/53029/60d04978Ef21f2d42/92baeb21f907cd24.jpg'

test('Image: render', async () => {
  const { container } = render(() => <Image src={url} width="100" height="100" showLoading />)
  const imgElement = container.querySelector('img')
  await fireEvent.load(imgElement)
  expect(container).toMatchSnapshot()
})

test('Image: load error', async () => {
  const { container } = render(() => <Image src='https://x' width="100" height="100" showError />)
  const imgElement = container.querySelector('img')
  await fireEvent.error(imgElement)
  expect(container.children[0].children[1].classList).toContain('nut-img-error')
  expect(container).toMatchSnapshot()
})

test('Image: loading', async () => {
  const { container } = render(() => <Image src='' width="100" height="100" showLoading />)
  expect(container.children[0].children[1].classList).toContain('nut-img-loading')
  expect(container).toMatchSnapshot()
})

test('Image: render Round', async () => {
  const { container } = render(() => <Image src={url} width="100" height="100" round />)
  const imgElement = container.querySelector('img')
  await fireEvent.load(imgElement)
  expect(container.children[0].classList).toContain('nut-image-round')
  expect(container).toMatchSnapshot()
})
