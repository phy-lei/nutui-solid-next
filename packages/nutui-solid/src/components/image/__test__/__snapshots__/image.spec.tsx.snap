// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Image: load error 1`] = `
<div>
  <div
    class="nut-image"
    style="width: 100px; height: 100px;"
  >
    <img
      alt=""
      class="nut-img"
      src="https://x"
      style="object-fit: fill; object-position: center;"
    />
    
    <div
      class="nut-img-error"
    >
      <svg
        aria-labelledby="imageError"
        class="nut-icon nut-icon-imageError "
        role="presentation"
        style="width: 16px; height: 20px;"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M696.43 465.25c40.61 0 73.53-32.72 73.53-73.09s-32.96-73.08-73.53-73.08-73.54 32.72-73.54 73.08 32.92 73.09 73.54 73.09zM535.74 650.62l-273.61-259.4a56.19 56.19 0 0 0-15.36 11.58l-114 119.63-6-222.77c-.73-26.82 20.84-49.41 47.67-50.14l201.77-5.44 47.34-50-250.43 6.75C119 202.33 76.33 247.17 77.78 301l7.32 271.4 4.5 166.78C91.06 793 136.05 835.47 190.2 834l176-4.75 12-.32 47.34-50 116.18-122.64z"
          fill="currentColor"
        />
        ,
        <path
          d="m861.71 214.8-242.83-33.86-55.09 41.46L855 263a49.05 49.05 0 0 1 41.79 55.13l-54.33 389.59-120.87-159.05A59.86 59.86 0 0 0 638.06 537l-44 33.08 65.07 86.15L603.84 698l-79.57 60.1-54.92 41.48 34.17 4.76 270.73 37.76a97.55 97.55 0 0 0 109.88-79.6 22.37 22.37 0 0 0 .65-3.44l60.53-434.16c7.44-53.32-29.95-102.62-83.6-110.1z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Image: loading 1`] = `
<div>
  <div
    class="nut-image"
    style="width: 100px; height: 100px;"
  >
    <img
      alt=""
      class="nut-img"
      src=""
      style="object-fit: fill; object-position: center;"
    />
    <div
      class="nut-img-loading"
    >
      <svg
        aria-labelledby="image"
        class="nut-icon nut-icon-image "
        role="presentation"
        style="width: 16px; height: 20px;"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M384 341.33a85.33 85.33 0 1 1-170.67 0 85.33 85.33 0 0 1 170.67 0z"
          fill="currentColor"
        />
        ,
        <path
          d="M0 170.67a128 128 0 0 1 128-128h768a128 128 0 0 1 128 128v682.66a128 128 0 0 1-128 128H128a128 128 0 0 1-128-128V170.67zM128 128a42.67 42.67 0 0 0-42.67 42.67v405.84a383.38 383.38 0 0 1 128-21.84c71.89 0 139.16 19.75 196.68 54.12C492.46 473.96 641.05 384 810.67 384c44.37 0 87.32 6.17 128 17.66V170.67A42.67 42.67 0 0 0 896 128H128zM85.33 668.74v184.59A42.67 42.67 0 0 0 128 896h380.97C488.28 751.25 363.8 640 213.33 640c-45.8 0-89.19 10.3-128 28.74zM594.99 896H896a42.67 42.67 0 0 0 42.67-42.67V491.18a383.36 383.36 0 0 0-128-21.85c-141.91 0-265.81 76.97-332.31 191.45A382.98 382.98 0 0 1 594.99 896z"
          fill="currentColor"
        />
      </svg>
    </div>
    
  </div>
</div>
`;

exports[`Image: render 1`] = `
<div>
  <div
    class="nut-image"
    style="width: 100px; height: 100px;"
  >
    <img
      alt=""
      class="nut-img"
      src="//img10.360buyimg.com/ling/jfs/t1/181258/24/10385/53029/60d04978Ef21f2d42/92baeb21f907cd24.jpg"
      style="object-fit: fill; object-position: center;"
    />
    
    
  </div>
</div>
`;

exports[`Image: render Round 1`] = `
<div>
  <div
    class="nut-image nut-image-round"
    style="width: 100px; height: 100px;"
  >
    <img
      alt=""
      class="nut-img"
      src="//img10.360buyimg.com/ling/jfs/t1/181258/24/10385/53029/60d04978Ef21f2d42/92baeb21f907cd24.jpg"
      style="object-fit: fill; object-position: center;"
    />
    
    
  </div>
</div>
`;
